pytradfri/resource.py

```diff
@@ -18,26 +18,27 @@
 class ApiResource:
     """Base object for resources returned from the gateway."""
 
-    def __init__(self, raw: ApiResourceResponse) -> None:
+    def __init__(self, raw: TypeRawList) -> None:
         """Initialize base object."""
-        self.raw = raw
+        self.raw: ApiResourceResponse = ApiResourceResponse(**raw)
```

Comment:
My understanding of the library is that this is how we populate the pydantic-model with data?

Reply:
Example:
```py
class ApiResource:
    """Base object for resources returned from the gateway."""

    _model_class: type[ResourceResponse]  = ResourceResponse
    raw: ResourceResponse

    def __init__(self, raw: TypeRaw) -> None:
        """Initialize base object."""
        self.raw = self._model_class(**raw)


class Device(ApiResource):
    """Base class for devices."""

    _model_class: type[DeviceResponse] = DeviceResponse
    raw: DeviceResponse
```

Reply:
Thank you for the pointers! I'll explore this further and see what I can come up with.

Reply:
If we don't want to update all modules in this PR we can add some conditional logic around the `raw` instantiation in the base resource class and set the `_model_class` as optional and keep the old way if `_model_class` is None.